zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -982,6 +991,75 @@ where
         }
         .boxed()
     }
+
+    fn z_list_unified_receivers(
+        &self,
+        address: String,
+    ) -> BoxFuture<Result<unified_address::Response>> {
+        use zcash_address::unified::Container;
+
+        async move {
+            let unified_address_data: (zcash_address::Network, zcash_address::unified::Address) =
+                zcash_address::unified::Encoding::decode(address.clone().as_str()).map_err(
+                    |error| Error {
+                        code: ErrorCode::ServerError(0),
+                        message: error.to_string(),
+                        data: None,
+                    },
+                )?;
+
+            let mut p2pkh = String::new();
+            let mut p2sh = String::new();
+            let mut orchard = String::new();
+            let mut sapling = String::new();
+
+            for item in unified_address_data.1.items() {
+                match item {
+                    zcash_address::unified::Receiver::Orchard(_data) => {
+                        if let Ok(addr) =
+                            zcash_address::unified::Address::try_from_items(vec![item])
+                        {
+                            orchard = addr.encode(&unified_address_data.0);
+                        }
```

Comment:
Address errors must return a RPC failure. Or if they are impossible they can be handled using `expect()`.

This could be avoided by the refactor I suggested, or we can add an error handler for each of the 4 address types.